code/game/machinery/newscaster.dm

```diff
@@ -702,6 +702,7 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 						var/obj/item/weapon/photo/P = photo
 						newMsg.img = P.img
 						newMsg.img_info = P.info
+						assassination_check(P)
```

Comment:
If I were a hardass I would ask you to use an event here, and to detach the `assassination_objectives` list from the newscaster. I would be just slightly more work for something a lot better code-wise.

Reply:
The assassination_objectives list is global though.

Reply:
Events just make this complicated honestly, I'd say this is fine.